### PR TITLE
Named dim override for Stable Diffusion 

### PR DESCRIPTION
**Pull request by linnealovespie**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by linnealovespie** [comment]
Refreshed this PR because it was getting messy

File: examples/stable_diffusion/config.py
Comment (jambayk) [inline examples/stable_diffusion/config.py:1]: i think this was supposed to be moved to sd_utils and not deleted?

Comment (linnealovespie) [reply examples/stable_diffusion/config.py:1]: Whoops you're right, thanks for pointing that out.

File: examples/stable_diffusion/config_vae_decoder.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
         "io_config": {
             "input_names": [ "latent_sample", "return_dict" ],
             "output_names": [ "sample" ],
-            "dynamic_axes": { "latent_sample": { "0": "batch", "1": "channels", "2": "height", "3": "width" } }
+            "dynamic_axes": { "latent_sample": { "0": "decoder_batch", "1": "decoder_channels", "2": "decoder_height", "3": "decoder_width" } }
```
Comment (jambayk) [inline examples/stable_diffusion/config_vae_decoder.json:10]: The linter is complaining about the formatting for these changes. 

Instructions for running the liner and applying the changes can be found at https://github.com/microsoft/Olive/blob/main/CONTRIBUTING.md#linting

**Review by jambayk — APPROVED** [review]
Thanks!

**Review by fdwr — APPROVED** [review]

**Pull request merged**